src/service/extension-location.js

```diff
@@ -105,10 +115,37 @@ export function calculateEntryPointScriptUrl(
 export function parseExtensionUrl(scriptUrl) {
   // Note that the "(\.max)?" group only applies to local dev.
   const matches = scriptUrl.match(
-    /^(.*)\/(.*)-([0-9.]+|latest)(\.max)?\.(?:js|mjs)$/i
+    /^(.*)\/(.*)-([0-9.]+|latest)(\.max)?\.(?:js|mjs)(\?.*)?$/i
   );
   return {
     extensionId: matches ? matches[2] : undefined,
     extensionVersion: matches ? matches[3] : undefined,
   };
 }
+
+/**
+ * Based on mode, determine which file extension to use for the script URL.
+ * @param {boolean=} opt_isLocalDev
+ * @return {string}
+ */
+function calculateFileExtension(opt_isLocalDev) {
+  if (getMode().sxg && opt_isLocalDev) {
+    return '.sxg.js';
+  } else if (getMode().sxg || getMode().esm) {
+    return '.mjs';
+  }
+  return '.js';
+}
+
+/**
+ * Add additional values to the script URL depending on mode, such as query parameters.
+ * @param {string} scriptUrl
+ * @param {boolean=} opt_isLocalDev
+ * @return {string}
+ */
+function maybeAddQueryParams(scriptUrl, opt_isLocalDev) {
+  // If opt_isLocalDev is true, then .sxg.js is already appended and thus we don't need to add a query param.
+  return getMode().sxg && !opt_isLocalDev
+    ? addParamToUrl(scriptUrl, 'f', 'sxg')
```

Comment:
Nit: This is expensive for URL's that we're already sure of. Can se just do `${scriptUrl}?f=sxg`?

Reply:
We could, but that would assume that there are no other query params.
This is the safest approach because it takes care of other query params.
But I guess we could do a more complex conditional check and modify the string that way?